### PR TITLE
SNOW-879602: Simplify github workflows

### PR DESCRIPTION
**Pull request by sfc-gh-dprzybysz**
### Description
Simplify github workflows to run tests on each cloud provider (AWS, AZURE, GCP), operating system (Centos, Mac, Windows), main supported node version (14.x, 16.x, 18.x - but Centos only with node 14.x).
Code coverage is aggregated from all Mac builds

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: ci/container/test_component.sh [hunk 1]
```diff
@@ -67,7 +67,7 @@ env | grep SNOWFLAKE_ | grep -v PASS
 echo "[INFO] Starting hang_webserver.py 12345"
 python3 $THIS_DIR/hang_webserver.py 12345 > hang_webserver.out 2>&1 &
 
-if [[ "$SHOULD_GENERATE_COVERAGE_REPORT" -eq "1" && "$CLOUD_PROVIDER" == "AWS" ]];
+if [[ "$SHOULD_GENERATE_COVERAGE_REPORT" -eq "1" ]];
```
Comment (sfc-gh-pfus) [inline ci/container/test_component.sh:70]: Does it work? I believe that `-eq` is used to compare numbers, for strings `==` should be used?

Comment (sfc-gh-dprzybysz) [reply ci/container/test_component.sh:70]: Yes, it works - for Mac we are gathering the coverage in tests (https://github.com/snowflakedb/snowflake-connector-nodejs/actions/runs/5830408639/job/15812373008?pr=612) and for linux is not gathered (https://github.com/snowflakedb/snowflake-connector-nodejs/actions/runs/5830408639/job/15812372847?pr=612) - in both links look at the `Tests` step

Comment (sfc-gh-pfus) [reply ci/container/test_component.sh:70]: Can we change it to be better POSIX ready?

**Review by sfc-gh-pfus — APPROVED** [review]

File: ci/container/test_component.sh (hunk 1 above)
Comment (sfc-gh-dprzybysz) [reply ci/container/test_component.sh]: replaced with `==`

**Pull request merged**